woocommerce/woocommerce
# Update `wp-env` to `8.2.0`

### Submission Review Guidelines:

-   I have followed the [WooCommerce Contributing Guidelines](https://github.com/woocommerce/woocommerce/blob/trunk/.github/CONTRIBUTING.md) and the [WordPress Coding Standards](https://make.wordpress.org/core/handbook/best-practices/coding-standards/).
-   I have checked to ensure there aren't other open [Pull Requests](https://github.com/woocommerce/woocommerce/pulls) for the same update/change.
-   I have reviewed my code for [security best practices](https://developer.wordpress.org/apis/security/).
-   Following the above guidelines will result in quick merges and clear and detailed feedback when appropriate.



### Changes proposed in this Pull Request:





Closes #38934.

This PR updates `wp-env` to `8.2.0` in order to make the _"Test with PHP 8.1"_ job in the _"Smoke test release"_ workflow pass again.

The version update was done by performing the following steps:
1. Run `pnpm i` to install dependencies which includes syncpack.
2. Manually updated `pinVersion` of `@wordpress/env` in `.syncpackrc`
3. Run `pnpm syncpack fix-mismatches`
4. Again run `pnpm i` to update the pnpm lock file.

Thanks so much @woocommerce/atlas for providing the steps above. I'm adding them too as reviewers to make sure no unwanted side effects to other packages were caused by this update.



### How to test the changes in this Pull Request:



Using the [WooCommerce Testing Instructions Guide](https://github.com/woocommerce/woocommerce/wiki/Writing-high-quality-testing-instructions), include your detailed testing instructions:

1. Checkout this branch.
    ```bash
    git checkout update/wp-env-820
    ```
6. Launch `wp-env` test environment with PHP 8.1 by running these commands from the monorepo root:
    ```bash
    export WP_ENV_PHP_VERSION='8.1' 
    pnpm install
    pnpm run build --filter=woocommerce
    cd plugins/woocommerce
    pnpm env:test
    ```
7. Verify that http://localhost:8086 was launched successfully.
8. Login to the WP Admin Dashboard and navigate to WooCommerce > Status. Verify that the PHP version is indeed 8.1.
1. Now, test for PHP 8.2 by running these commands, again from the monorepo root:
    ```bash
    cd plugins/woocommerce
    export WP_ENV_PHP_VERSION='8.2'
    pnpm env:restart
    ```
    When asked if you want to _"remove Docker containers, volumes, networks, and images associated with the WordPress instance"_, answer _"Yes"_.
1. Again, verify that the local testing environment was launched successfully.
1. Login, go to WooCommerce > Status, and verify that PHP version is now 8.2.



### Changelog entry




<details>

#### Significance




#### Type




#### Message 

#### Comment 

</details>
